be1-go/channel/consensus/mod.go

```diff
@@ -123,10 +147,88 @@ func (c *Channel) NewConsensusRegistry() registry.MessageRegistry {
 	registry.Register(messagedata.ConsensusPropose{}, c.processConsensusPropose)
 	registry.Register(messagedata.ConsensusAccept{}, c.processConsensusAccept)
 	registry.Register(messagedata.ConsensusLearn{}, c.processConsensusLearn)
+	registry.Register(messagedata.ConsensusFailure{}, c.processConsensusFailure)
 
 	return registry
 }
 
+// timeoutFailure sends a failure message during a timeout
+func (c *Channel) timeoutFailure(instance *ConsensusInstance, messageID string) {
+	byteMsg, err := c.createFailureMessage(instance, messageID)
+	if err != nil {
+		c.log.Err(err).Msg(failedFailureCreation)
+		return
+	}
+
+	err = c.publishNewMessage(byteMsg)
+	if err != nil {
+		c.log.Err(err).Msg(failedFailureSending)
+	}
+}
+
+// startTimer starts the timeout logic for the consensus
+func (c *Channel) startTimer(instance *ConsensusInstance, messageID string) {
+
+	timeoutChan := instance.electInstances[messageID].timeoutChan
+
+	for {
+		select {
+		case action := <-timeoutChan:
+			switch action {
+			// Stop the timer when receiving one action finishing it
+			case messagedata.ConsensusActionLearn,
```

Comment:
I can't remove it because I don't want to return if the channel receive for exemple messagedata.ConsensusElectAccept. I just want to start the timer from zero in this case.
But I will throw an error if the received string doesn't correspond with anything